task/host-sp-comms/src/bsp/gimlet_bcd.rs

```diff
@@ -0,0 +1,674 @@
+    ///
+    /// Returns a designator (e.g. J206) and constructor function
+    fn get_sharkfin_vpd(i: usize) -> ([u8; 4], fn(TaskId) -> I2cDevice) {
+        let fs = [
+            i2c_config::devices::at24csw080_sharkfin_a_vpd,
+            i2c_config::devices::at24csw080_sharkfin_b_vpd,
+            i2c_config::devices::at24csw080_sharkfin_c_vpd,
+            i2c_config::devices::at24csw080_sharkfin_d_vpd,
+            i2c_config::devices::at24csw080_sharkfin_e_vpd,
+            i2c_config::devices::at24csw080_sharkfin_f_vpd,
+            i2c_config::devices::at24csw080_sharkfin_g_vpd,
+            i2c_config::devices::at24csw080_sharkfin_h_vpd,
+            i2c_config::devices::at24csw080_sharkfin_i_vpd,
+            i2c_config::devices::at24csw080_sharkfin_j_vpd,
+        ];
+        // The base name is J206, so we count up from there
+        let mut name = *b"J2__";
+        name[2] = ((i + 6) / 10) as u8 + b'0';
+        name[3] = ((i + 6) % 10) as u8 + b'0';
+        (name, fs[i])
+    }
+
+    fn dimm_inventory_lookup(&mut self, sequence: u64, index: u32) {
+        // Build a name of the form `m{index}`, to match the designator
+        let mut name = [0; 32];
+        name[0] = b'm';
+        if index >= 10 {
+            name[1] = b'0' + (index / 10) as u8;
+            name[2] = b'0' + (index % 10) as u8;
+        } else {
+            name[1] = b'0' + index as u8;
+        }
+
+        let packrat = &self.packrat; // partial borrow
+        let mut data = InventoryData::DimmSpd([0u8; 512]);
+        self.tx_buf.try_encode_inventory(sequence, &name, || {
+            // TODO: does packrat index match PCA designator?
+            if packrat.get_spd_present(index as usize) {
+                let InventoryData::DimmSpd(out) = &mut data
+                    else { unreachable!(); };
+                packrat.get_full_spd_data(index as usize, out);
+                Ok(&data)
+            } else {
+                Err(InventoryDataResult::DeviceAbsent)
+            }
+        });
+    }
+
+    /// Reads the 128-byte unique ID from an AT24CSW080 EEPROM
+    ///
+    /// `data` is passed in to reduce stack frame size, since we already require
+    /// an allocation for it on the caller's stack frame.
+    fn read_at24csw080_id(
+        &mut self,
+        sequence: u64,
+        name: &[u8],
+        f: fn(userlib::TaskId) -> I2cDevice,
+        data: &mut InventoryData,
+    ) {
+        // This should be done by the caller, but let's make it obviously
+        // correct (since we destructure it below).
+        *data = InventoryData::At24csw08xSerial([0u8; 16]);
+        let dev = At24Csw080::new(f(I2C.get_task_id()));
+        self.tx_buf.try_encode_inventory(sequence, name, || {
+            let InventoryData::At24csw08xSerial(id) = data
+                else { unreachable!(); };
+            for (i, b) in id.iter_mut().enumerate() {
+                // TODO: add an API to make this a single IPC call?
```

Comment:
"do you want to address this before merging" check

Reply:
No, I think it's fine as-is